gpflow/kullback_leiblers.py

```diff
@@ -33,38 +33,46 @@ def gauss_kl(q_mu, q_sqrt, K=None):
     We assume N multiple independent distributions, given by the columns of
     q_mu and the last dimension of q_sqrt. Returns the sum of the divergences.
 
-    q_mu is a matrix (M x L), each column contains a mean.
+    q_mu is a matrix [M, L], each column contains a mean.
 
-    q_sqrt can be a 3D tensor (L x M x M), each matrix within is a lower
+    q_sqrt can be a 3D tensor [L, M, M], each matrix within is a lower
         triangular square-root matrix of the covariance of q.
-    q_sqrt can be a matrix (M x L), each column represents the diagonal of a
+    q_sqrt can be a matrix [M, L], each column represents the diagonal of a
         square-root matrix of the covariance of q.
 
-    K is the covariance of p.
-    It is a positive definite matrix (M x M) or a tensor of stacked such matrices (L x M x M)
-    If K is None, compute the KL divergence to p(x) = N(0, I) instead.
+    K is the covariance of p, [M, M] or [L, M, M]
+    K_cholesky is the cholesky of the covariance of p, [M, M] or [L, M, M]
+    Either `K` or `K_cholesky` have to be None.
+    If `K` and `K_cholesky` are None, compute the KL divergence to p(x) = N(0, I) instead.
```

Comment:
It would be good to make a bit clearer what happens. How about:

"If no K matrix is given (both `K` and `K_cholesky` are None), computes the KL divergence from p(x) = N(0, I) instead.
The K matrix can be passed either directly as `K`, or as its cholesky factor, `K_cholesky`. In either case, it can be a single matrix [M, M], in which case the sum of the L KL divergences is computed by broadcasting, or L different covariances [L, M, M]."

(note that it is the KL[q || p] *from* p *to* q)